src/lib/util/connection-util.ts

```diff
@@ -0,0 +1,81 @@
+/**
+ * Used to infer the parameters of a callback function.
+ */
+export type InferSignalParameters<S> = S extends SignalLike
+	? Parameters<
+			Parameters<
+				S["Connect"] extends Callback
+					? S["Connect"]
+					: S["connect"] extends Callback
+					? S["connect"]
+					: S["on"] extends Callback
+					? S["on"]
+					: S["do"] extends Callback
+					? S["do"]
+					: never
+			>[0]
+	  >
+	: never;
```

Comment:
Remove the lowercase connect ones, haven't really seen that stuff in 5-ish years so I'd rather not support bad code